server/me.go

```diff
@@ -74,6 +78,93 @@ func getPrincipal(ctx context.Context) (oauth2.Principal, error) {
 	return principal, nil
 }
 
+func getValidPrincipal(ctx context.Context) (oauth2.Principal, error) {
+	p, err := getPrincipal(ctx)
+	if err != nil {
+		return p, err
+	}
+	if p.Subject == "" {
+		return oauth2.Principal{}, fmt.Errorf("Token not found")
+	}
+	if p.Issuer == "" {
+		return oauth2.Principal{}, fmt.Errorf("Token not found")
+	}
+	return p, nil
+}
+
+type meOrganizationRequest struct {
+	OrganizationID string `json:"currentOrganization"`
+}
+
+// MeOrganization changes the user's current organization on the JWT and responds
+// with the same semantics as Me
+func (s *Service) MeOrganization(auth oauth2.Authenticator) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		principal, err := auth.Validate(ctx, r)
+		if err != nil {
+			s.Logger.Error("Invalid principal")
+			w.WriteHeader(http.StatusForbidden)
+			return
+		}
+		var req meOrganizationRequest
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+			invalidJSON(w, s.Logger)
+			return
+		}
+
+		// validate that the organization exists
+		orgID, err := strconv.ParseUint(req.OrganizationID, 10, 64)
+		if err != nil {
+			Error(w, http.StatusInternalServerError, err.Error(), s.Logger)
+			return
+		}
+		_, err = s.OrganizationsStore.Get(ctx, chronograf.OrganizationQuery{ID: &orgID})
+		if err != nil {
+			Error(w, http.StatusBadRequest, err.Error(), s.Logger)
+			return
+		}
+
+		p, err := getValidPrincipal(ctx)
+		if err != nil {
+			invalidData(w, err, s.Logger)
+			return
+		}
+		scheme, err := getScheme(ctx)
+		if err != nil {
+			invalidData(w, err, s.Logger)
+			return
+		}
+		// validate that user belongs to organization
+		ctx = context.WithValue(ctx, "organizationID", req.OrganizationID)
+		_, err = s.OrganizationUsersStore.Get(ctx, chronograf.UserQuery{
```

Comment:
This could be removed when you have new logic for creating filtered stores based on org.